### PR TITLE
Address warnings generated when compiled with gcc 4.8 and -Wall

### PR DESCRIPTION
**Pull request by rkimballn1**
* Change NGRAPH_WARNINGS_AS_ERRORS to not check an environment variable and only use the parameter passed to cmake
* Alphabetize the dump of flag settings, making it much easier to find a flag
* Fix warnings that are new due to -Wall and gcc 4.8

### CODE REVIEW

**Comment by silee2** [comment]
Seems like Werror is passed to Windows as well : `34>cl : Command line error D8021: invalid numeric argument '/Werror' [D:\jenkins\workspace\WORKSPACE_1\ngraph\build\src\ngraph\ngraph.vcxproj]`
`C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\CL.exe /c /ID:\jenkins\workspace\WORKSPACE_1\ngraph\src\ngraph /ID:\jenkins\workspace\WORKSPACE_1\ngraph\src /ID:\jenkins\workspace\WORKSPACE_1\ngraph\build\json\src\ext_json\include /nologo /W0 /WX- /diagnostics:classic /MP /O2 /Ob2 /D WIN32 /D _WINDOWS /D EIGEN_MPL2_ONLY /D TBB_USE_THREADING_TOOLS /D NOMINMAX /D _CRT_SECURE_NO_WARNINGS /D NGRAPH_CPU_ENABLE /D NGRAPH_DYNAMIC_COMPONENTS_ENABLE /D NDEBUG /D "SHARED_LIB_PREFIX=\"\"" /D "SHARED_LIB_SUFFIX=\".dll\"" /D "NGRAPH_VERSION=\"0.28.0-rc.0+6990de8\"" /D "PROJECT_ROOT_DIR=\"D:/jenkins/workspace/WORKSPACE_1/ngraph\"" /D IN_NGRAPH_LIBRARY /D "CMAKE_INTDIR=\"Release\"" /D ngraph_EXPORTS /D _WINDLL /D _MBCS /Gm- /EHsc /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /Fo"ngraph.dir\Release\\" /Fd"ngraph.dir\Release\vc141.pdb" /Gd /TP /wd4251 /wd4275 /errorReport:queue  /bigobj -Werror`

**Comment by silee2** [comment]
onnx_import also show some issue: `/home/dockuser/ngraph/src/ngraph/frontend/onnx_import/core/attribute.hpp:222:28: error: enumeration value 'AttributeProto_AttributeType_UNDEFINED' not handled in switch [-Werror=switch-enum]
                     switch (attribute.type())`

File: src/ngraph/runtime/cpu/builder/matmul_bias.cpp [hunk 1]
```diff
@@ -87,6 +87,8 @@ namespace ngraph
                                    out0_buffer_index,
                                    element_type](CPURuntimeContext* ctx,
                                                  CPUExecutionContext* /* ectx */) {
+#pragma GCC diagnostic push
```
Comment (diyessi) [inline src/ngraph/runtime/cpu/builder/matmul_bias.cpp]: These need to be fixed for gcc 4.8

File: CMakeLists.txt [hunk 2]
```diff
@@ -407,10 +414,8 @@ if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "^(Apple)?Clang$" AND NOT NGRAPH_BUILD_DI
     include( cmake/clang_4_0_flags.cmake )
 endif()
 
-ngraph_var(NGRAPH_WARNINGS_AS_ERRORS DEFAULT "OFF")
 if (${NGRAPH_WARNINGS_AS_ERRORS})
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
```
Comment (silee2) [inline CMakeLists.txt]: `Werror` is not a valid option for MSVC
CI is failing windows build.
```cl : Command line error D8021: invalid numeric argument '/Werror' [D:\jenkins\workspace\WORKSPACE_2@2\ngraph\build\src\ngraph\ngraph.vcxproj]```

**Pull request merged**